src/common/procedure-test/src/lib.rs

```diff
@@ -52,3 +64,53 @@ pub async fn execute_procedure_until_done(procedure: &mut BoxedProcedure) {
         }
     }
 }
+
+/// Executes a procedure once.
+///
+/// Returns whether the procedure is done.
+pub async fn execute_procedure_once(
+    procedure_id: ProcedureId,
+    provider: MockContextProvider,
+    procedure: &mut dyn Procedure,
+) -> bool {
+    let ctx = Context {
+        procedure_id,
+        provider: Arc::new(provider),
+    };
+
+    match procedure.execute(&ctx).await.unwrap() {
+        Status::Executing { .. } => false,
+        Status::Suspended { subprocedures, .. } => {
+            assert!(
+                subprocedures.is_empty(),
+                "Executing subprocedure is unsupported"
+            );
+            false
+        }
+        Status::Done => true,
+    }
+}
+
+/// Executes a procedure until it returns [Status::Suspended] or [Status::Done].
+///
+/// Returns `Some` if it returns [Status::Suspended] or `None` if it returns [Status::Done].
+pub async fn execute_parent_procedure(
```

Comment:
The function name is confusing. But I can't find a better name.

Reply:
What about execute_until_suspended_or_done?

Reply:
> What about execute_until_suspended_or_done?

I think you mean "poll".

Reply:
`poll` sounds like `execute once`.